### PR TITLE
Added mongodb.

### PR DESCRIPTION
**Pull request by justinasalt**

### CODE REVIEW

File: app/lib/controllers/crmController.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import * as mongoose from "mongoose";
+import { ContactSchema } from "../models/crmModel";
+import { Request, Response } from "express";
+
+const Contact = mongoose.model("Contact", ContactSchema);
+
+export class ContactController {
+  public addNewContact(req: Request, res: Response): void {
+    const newContact = new Contact({
+      firstName: "dik",
```
Comment (martynaspuskunigis) [inline app/lib/controllers/crmController.ts]: LOL

File: app/lib/controllers/crmController.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      lastName: "sex"
```
Comment (martynaspuskunigis) [inline app/lib/controllers/crmController.ts]: pls don't delete comments, thank you

File: app/lib/app.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import * as express from "express";
+import * as bodyParser from "body-parser";
+import * as mongoose from "mongoose";
+import { Routes } from "./routes/DbRoutes";
```
Comment (DonatasBrazionis) [inline app/lib/app.ts]: Imports spacing.

File: app/lib/app.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+class App {
+
+    public app: express.Application;
+    public routePrv: Routes = new Routes();
+    public mongoUrl: string = "mongodb+srv://justinasalt:martynasgrazuolis@cluster0-x9opy.mongodb.net/test?retryWrites=true";
```
Comment (martynaspuskunigis) [inline app/lib/app.ts]: I shouldn't be seeing this.

File: app/lib/controllers/actorsDbController.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import * as mongoose from "mongoose";
+import { Request, Response } from "express";
+
+import { ActorSchema } from "../models/actorDbModel";
+
+const Actors = mongoose.model("actors", ActorSchema);
+
+export class ActorController {
+  public addNewActor(req: Request, res: Response): void {
+    const newMovie = new Actors(req.body);
```
Comment (martynaspuskunigis) [inline app/lib/controllers/actorsDbController.ts]: new newMovie = new Actors? fix naming.

File: app/lib/controllers/actorsDbController.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    newMovie.save((err, movie) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(movie);
+    });
+  }
+  public getAllActors(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/actorsDbController.ts]: Before every method add new line. And do not need after `export class ActorController {`.

File: app/lib/controllers/videoDbController.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import * as mongoose from "mongoose";
+import { Request, Response } from "express";
+
+import { VideoSchema } from "../models/videoDbModel";
+
+const Videos = mongoose.model("videos", VideoSchema);
+
+export class VideoController {
+  public addNewVideo(req: Request, res: Response): void {
+    const newMovie = new Videos(req.body);
```
Comment (martynaspuskunigis) [inline app/lib/controllers/videoDbController.ts]: I think Movie and Video are different things in your project, fix naming.

File: app/lib/controllers/actorsDbController.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    Actors.find({}, (err, movies) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(movies);
+    });
+  }
+
+  public getActorWithID(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/actorsDbController.ts]: `getActorWithID` -> `getActorWithId`.

File: app/lib/controllers/movieDbController.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  public addNewMovie(req: Request, res: Response): void {
+    const newMovie = new Movies(req.body);
+    newMovie.save((err, movie) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(movie);
+    });
+  }
+  public getAllMovies(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/movieDbController.ts]: Before every method add new line. And do not need after `export class MovieController {`.

File: app/lib/controllers/movieDbController.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  public searchMovies(req: Request, res: Response): void {
+    Movies.find({}, (err, movies) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(searchMovie(movies, req.params.query));
+    });
+  }
+
+  public getMovieWithID(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/movieDbController.ts]: `getMovieWithID` -> `getMovieWithId`.

File: app/lib/models/videoDbModel.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import * as mongoose from "mongoose";
+
+const Schema = mongoose.Schema;
+
+export const VideoSchema = new Schema({
+  movie_id: { type: Number },
```
Comment (martynaspuskunigis) [inline app/lib/models/videoDbModel.ts]: Why not camelCase?

File: app/lib/controllers/videoDbController.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    newMovie.save((err, movie) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(movie);
+    });
+  }
+  public getAllVideos(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/videoDbController.ts]: Before every method add new line. And do not need after `export class VideoController {`.

File: app/lib/controllers/videoDbController.ts [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    Videos.find({}, (err, movies) => {
+      if (err) {
+        res.send(err);
+      }
+      res.json(movies);
+    });
+  }
+
+  public getVideoWithID(req: Request, res: Response): void {
```
Comment (DonatasBrazionis) [inline app/lib/controllers/videoDbController.ts]: `getVideoWithID` -> `getVideoWithId`.

File: package.json [hunk 14]
```diff
@@ -4,15 +4,20 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "tsc": "tsc"
+    "build": "tsc",
+    "dev": "ts-node ./app/lib/server.ts",
```
Comment (martynaspuskunigis) [inline package.json]: Don't use ts-node, use nodemon instead.

File: app/lib/routes/DbRoutes.ts [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    });
+
+    app.route("/all-actors").get(this.actorController.getAllActors);
+    app.route("/all-actors").post(this.actorController.addNewActor);
+
+    app.route("/all-actors/:actorID").get(this.actorController.getActorWithID);
+    app.route("/all-actors/:actorID").get(this.actorController.getActorWithID);
+    app.route("/").get((req: Request, res: Response) => {
+      res.status(200).send({
+        message: "GET request successfulll!!!!"
```
Comment (DonatasBrazionis) [inline app/lib/routes/DbRoutes.ts]: There is no such word  `successfulll`.
Why so much exclamation marks?

File: app/lib/app.ts (hunk 4 above)
Comment (justinasalt) [reply app/lib/app.ts]: You won't from now on.

File: app/lib/models/videoDbModel.ts (hunk 11 above)
Comment (justinasalt) [reply app/lib/models/videoDbModel.ts]: Can't write numbers in camel case.

File: package.json (hunk 14 above)
Comment (justinasalt) [reply package.json]: Fixed

File: app/lib/routes/DbRoutes.ts (hunk 15 above)
Comment (justinasalt) [reply app/lib/routes/DbRoutes.ts]: Sorry,  I overreacted :)

**Comment by martynaspuskunigis** [comment]
Good luck with the conflicts :)

**Pull request merged**